### PR TITLE
fix(dialog): fix scrollbar pointer-down click prevent dialog closing

### LINKED ISSUE #624: [Bug]: Prevent dismissing Dialog when interacting with Toasts (Overlay Components) or native scrollbar
### Environment

```bash
Radix Vue version: 1.3.2
Vue version: 3.4
```

### Link to minimal reproduction

https://stackblitz.com/edit/t2zzma?file=src%2FApp.vue

Discord Chat:

https://discord.com/channels/1121805548782878813/1194995659758120971

### Steps to reproduce

https://github.com/radix-vue/shadcn-vue/assets/17789047/87dbfc3f-a0db-488f-9ebc-b37553012266

### Describe the bug

Dialog Component will dismiss want to interact with Toasts or native scrollbar 

### Expected behavior

Dialog must remain open, and it should be dismissed when I click on `DialogOverlay`

### PR DESCRIPTION
**Pull request by romanhrynevych**
this PR closes #624

### CODE REVIEW

**Comment by zernonia** [comment]
Thanks for the PR @romanhrynevych . we can prevent dismissing the dialog when clicking on the Toast. So we dont need explicit fix for this. 

Your fixes for clicking on scrollbar looks good to me 😁

**Comment by zernonia** [comment]
> @zernonia thank you for your great job on this package ❤️
> 
> How can we prevent closing Dialog by dismissing it? I added video asset to clarify bug that I'm cannot fix for now)
>  

We cant prevent default following this comment [here](https://github.com/radix-ui/primitives/issues/386#issuecomment-1044509476). 

The above shows `Toast` component, but it's technically the same for `sonner` component

**Comment by romanhrynevych** [comment]
Think that it will be helpful to point this inside docs, so I add some demo for this type of usage, maybe this will clarify things for more devs 🙂 


https://github.com/radix-vue/radix-vue/assets/82209198/c5cecb2d-e7a8-472f-99f8-2fd9e127134a

@sadeghbarati hi 👋  What you think about this?)

File: docs/package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "@stackblitz/sdk": "^1.9.0",
     "@vueuse/core": "^10.5.0",
     "radix-vue": "link:../packages/radix-vue",
-    "vue": "^3.3.8"
+    "vue": "^3.3.8",
+    "vue-sonner": "^1.0.2"
```
Comment (zernonia) [inline docs/package.json:17]: Let's not use `vue-sonner` as an example. Instead let's use the `Toast` componennt

**Comment by sadeghbarati** [comment]
Great Job @romanhrynevych ❤️

File: docs/package.json (hunk 1 above)
Comment (romanhrynevych) [reply docs/package.json:17]: Ok, I will change it and give you a response ❤️

**Comment by romanhrynevych** [comment]
I just noticed one more issue, need to resolve it too.

So, when I add Toast component, because of it position (on top of our Dialog) we never will click on toast because `DialogOverlay` overlaps it, did you know how to `Teleport` Toast to the end of `body`? I search inside `vue` docs, but nothing 😞 


https://github.com/radix-vue/radix-vue/assets/82209198/5a68e00d-7ca4-4549-b52a-17b6e8def107

Also, I don't see any need to `@interact-outside` call on `DialogContent`, because it will never called when clicking on another elements 🤔

**Comment by zernonia** [comment]
Thanks @romanhrynevych ! After setting `pointer-events-auto` to `<ToastViewport>`, we can click on the Toast without closing the Dialog. I will revert the changes to use `sonner` as I believe other dev would be using other Toast package, and it demonstrates a good example of `@interactOutside`!

**Comment by zernonia** [comment]
Thanks for the PR @romanhrynevych ! Well done! 💚

**Pull request merged**